reactivesocket-core/src/main/java/io/reactivesocket/Frame.java

```diff
@@ -397,6 +397,11 @@ public static int requestN(final Frame frame) {
     public static class Request {
         private Request() {}
 
+        public static Frame from(int streamId, FrameType type, Payload payload, long initialRequestN) {
+            int v = initialRequestN > Integer.MAX_VALUE ? Integer.MAX_VALUE : (int) initialRequestN;
```

Comment:
Can we reduce initialRequestN to int now?  Also this seems unsafe for input like -1, as that will get past this check.